benchflow-experiment-manager/application/src/main/java/cloud/benchflow/experimentmanager/scheduler/ExperimentTaskScheduler.java

```diff
@@ -15,43 +15,50 @@
 import cloud.benchflow.experimentmanager.tasks.running.ReExecuteTrialTask;
 import cloud.benchflow.experimentmanager.tasks.running.execute.ExecuteTrial.TrialStatus;
 import cloud.benchflow.experimentmanager.tasks.start.StartTask;
-
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Future;
-
+import java.util.concurrent.LinkedBlockingDeque;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
  * @author Jesper Findahl (jesper.findahl@usi.ch) created on 2017-04-19
  */
-public class ExperimentTaskController {
+public class ExperimentTaskScheduler {
 
   private static Logger logger =
-      LoggerFactory.getLogger(ExperimentTaskController.class.getSimpleName());
+      LoggerFactory.getLogger(ExperimentTaskScheduler.class.getSimpleName());
 
   private ConcurrentMap<String, Future> experimentTasks = new ConcurrentHashMap<>();
 
-  // TODO - running queue (1 element)
+  // ready queue
+  private BlockingQueue<String> readyQueue = new LinkedBlockingDeque<>();
 
-  // TODO - ready queue
+  // running queue (1 element)
+  private BlockingQueue<String> runningQueue = new ArrayBlockingQueue<>(1, true);
 
   private BenchFlowExperimentModelDAO experimentModelDAO;
   private TrialModelDAO trialModelDAO;
   // TODO - should go into a stateless queue (so that we can recover)
   private ExecutorService experimentTaskExecutorService;
   private BenchFlowTestManagerService testManagerService;
 
-  public ExperimentTaskController(ExecutorService experimentTaskExecutorService) {
+  public ExperimentTaskScheduler(ExecutorService experimentTaskExecutorService) {
 
     this.experimentTaskExecutorService = experimentTaskExecutorService;
 
     this.experimentModelDAO = BenchFlowExperimentManagerApplication.getExperimentModelDAO();
     this.trialModelDAO = BenchFlowExperimentManagerApplication.getTrialModelDAO();
     this.testManagerService = BenchFlowExperimentManagerApplication.getTestManagerService();
+
+    // start dispatcher in separate thread
+    new Thread(new ExperimentDispatcher(readyQueue, runningQueue)).start();
```

Comment:
I'm not so in favor of sharing data structures in that way since it creates strong coupling. Also we have to depend on the order of how objects are instantiated. I know I have that someplace else, but I would avoid it if possible. I would then prefer to call `initialize` from the main application.
In fact maybe all classes (except resources) that get instances from the main application should have an `initialize` method where these instances are assigned. Then it would be two clear steps in the main application:
- instantiate the objects
- call initialize on each instance to assign dependencies